src/methods/potential_theory/methods.jl

```diff
@@ -0,0 +1,465 @@
+function FugacityCoefficientMethod(prob::PTAProblem; kwargs...)
+    x0 = PTA_x0(prob)
+    FugacityCoefficientMethod(; x0=x0, kwargs...)
+end
+
+"""
+    solve_at_potential_fugacity(eos, P_init, z_init, T, z_bulk, f_bulk, P_bulk, ε_vec, alg; verbose=false)
+
+Solve for P(ε) and z(ε) at a single potential value using fugacity coefficient method
+with stability checks to avoid oscillation.
+
+Paper equations:
+- Composition: z^i(ε) = [z_g^i * f_g^i * P_g * exp(ε^i/RT)] / [f^i(P,z)]
+- Constraint:  Σ_i [z_g^i * f_g^i * P_g * exp(ε^i/RT)] / [f^i(P,z)] - 1 = 0
+
+Uses Clapeyron's lnϕ and ForwardDiff for automatic differentiation.
+
+Algorithm (with stability check from Clapeyron bubble pressure):
+OUTER LOOP (Newton on pressure):
+  1. Save checkpoint: z_restart, P_restart
+  2. INNER LOOP (successive substitution on composition):
+     a. Calculate f(P, z) using lnϕ
+     b. Update z from fugacity equation
+     c. Check convergence on z
+     d. STABILITY CHECK: ||z - z_bulk|| < tol_stability?
+        - If yes: restore checkpoint, exit inner loop, force P update
+        - If no: continue
+  3. Calculate pressure residual and derivative using ForwardDiff
+  4. Newton update on P with damping
+  5. Check convergence
+"""
+function solve_at_potential_fugacity(eos, P_init, z_init, T, z_bulk, f_bulk, P_bulk, ε_vec, alg; verbose=false)
+    
+    # Unpack
+    RT = Rg(eos) * T
+    n_comp = length(z_bulk)
+    
+    # Pre-compute constant term: C^i = z_g^i * f_g^i  * exp(ε^i/RT)
+    C = z_bulk .* f_bulk .* exp.(ε_vec ./ RT)
+    
+    # Tolerances
+    tol_z = alg.abstol
+    tol_P = alg.reltol
+    tol_stability = abs2(cbrt(tol_z))  # Stability threshold (Clapeyron pattern)
+    
+    # Initialize
+    P = P_init
+    z = copy(z_init)
+    
+    # OUTER LOOP: Newton iteration on pressure
+    for j in 1:alg.maxiter_outer
+        
+        # Checkpoint for stability restart
+        z_restart = copy(z)
+        P_restart = P
+        
+        valid_iter = true  # Flag for valid inner loop convergence
+        
+        # INNER LOOP: Successive substitution on composition
+        for i in 1:alg.maxiter_inner
+            
+            # Calculate fugacity coefficients at current (P, z)
+            # Using Clapeyron's lnϕ function
+            lnϕ_z = lnϕ(eos, P, T, z)[1]
+            ϕ_z = exp.(lnϕ_z)
+            f_z = ϕ_z .* P  # Fugacities
+            
+            # Update composition: z^i = C^i / f^i(P,z)
+            z_new = C ./ f_z
+            z_new = z_new / sum(z_new) # Normalize
+            
+            # Check convergence on composition
+            if norm(z_new - z, Inf) < tol_z
+                z = z_new
+                break
+            end
+            
+            #═════════════════════════════════════════════════════
+            # STABILITY CHECK (prevents oscillation/trivial solution)
+            #═════════════════════════════════════════════════════
+            stability = norm(z_new - z_bulk, Inf)  # ||z - z_bulk||∞
```

Comment:
a good way for checking if z converged to z_bulk is by using K-values
```julia
K = z/z_bulk
sum(abs2,log.(K))
```
or, in non-allocating terms:
```julia
function z_norm(z,w)
    z_norm = zero(Base.promote_eltype(z,w))
    for i in 1:length(z)
        z_norm += log(w[i]/z[i])^2
    end
    return z_norm
end
```
(we use that function in Clapeyron to check if two compositions are too similar)